bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/AbstractDiscoveryService.java

```diff
@@ -33,13 +36,12 @@
  * @author Oliver Libutzki - Initial contribution
  * @author Kai Kreuzer - Refactored API
  * @author Dennis Nobel - Added background discovery configuration through Configuration Admin
+ * @author Andre Fuechsel - Added removeOlderResults
  */
 public abstract class AbstractDiscoveryService implements DiscoveryService {
 
     private final Logger logger = LoggerFactory.getLogger(AbstractDiscoveryService.class);
 
-    static protected final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(3);
```

Comment:
In order to make it easy for users (and not to break backward compatibility), I would suggest to keep the scheduler field here and just initialize it with ```DiscoveryThreadPool.getScheduler()```